### PR TITLE
lldpd: update and add cflags to reduce size

### PR DESCRIPTION
**Pull request by lucize**
enable lto cflags to reduce package size
remove upstreamed patch
revork libcap hack patch as it was not working anymore

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>
Maintainer: @stintel 

size differences and tested on:

mips
new 114 461
old 129 991

kirkwood
new 127 603
old 139 291

mvebu
new 126 134
old 138 730

x86_64
new 130 050
old 144 171

### CODE REVIEW

File: package/network/services/lldpd/patches/001-disable_libcap.patch [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
- # Third-party libraries
- lldp_CHECK_LIBEVENT
--lldp_CHECK_LIBCAP
-+
-+#libcap is not a core package so disable this capability
-+#lldp_CHECK_LIBCAP
-+libcap_LIBS=
-+libcap_CFLAGS=
-+AC_SUBST([libcap_LIBS])
-+AC_SUBST([libcap_CFLAGS])
```
Comment (hauke) [inline package/network/services/lldpd/patches/001-disable_libcap.patch:14]: I would prefer if you do these changes in the ```configure.ac``` and then add this to the package Makefile to recreate the the ```configure``` file.
```PKG_FIXUP:=autoreconf```

**Comment by stintel** [comment]
Also: don't bump and make unrelated other changes in a single commit, this makes bisecting impossible. So please address Hauke's comment, and split the PR in 2 commits. Ideally, the patch to disable libcap should be submitted upstream. The current one was previously accepted by another developer without me noticing in time, but now that changes are required, please upstream first.

**Review by stintel — CHANGES_REQUESTED** [review]
Also: don't bump and make unrelated other changes in a single commit, this makes bisecting impossible. So please address Hauke's comment, and split the PR in 2 commits. Ideally, the patch to disable libcap should be submitted upstream. The current one was previously accepted by another developer without me noticing in time, but now that changes are required, please upstream first.

**Comment by lucize** [comment]
didn't had time to test the autoreconf, I'll approach upstream about libcap, but -flto can be very useful on 4MB devices

**Comment by stintel** [comment]
Another option is to move libcap from the packages feed to OpenWrt base and add the libcap dependency. I actually prefer this, as it would allow the monitor process to drop its privileges instead of running as root, which improves security. This will not help 4MB devices, but honestly I prefer security over keep supporting almost unsupportable devices.

**Comment by lucize** [comment]
yes, let's do that and if upstreamed we'll add option not to link against libcap

**Comment by stintel** [comment]
> Another option is to move libcap from the packages feed to OpenWrt base

@p-wassi any objections?

**Comment by stintel** [comment]
Prepared it here: https://git.openwrt.org/?p=openwrt/staging/stintel.git;a=shortlog;h=refs/heads/libcap

**Comment by hauke** [comment]
Adding libcap to the main repository is a good idea.

**Comment by stintel** [comment]
> Adding libcap to the main repository is a good idea.

@ynezz seems to disagree, see http://lists.openwrt.org/pipermail/openwrt-devel/2021-March/034286.html
I prefer we continue the discussion there.

**Pull request closed without merging**